Code optimization and cleanup

MeuhV1 opened the pull request:
Various code optimization in Plague and RFCUtils
Prevented 1650 plagues starting in 1700 AD scenario

Code Cleanup:
Changed various pCurrent into pPlot (This makes it more obvious that it deals with plots)
Removed isNoVassal() function (just use not isAVassal() function)
Removed tabs and spaces at the end of lines, replaces spaces with tabs for consistency

MeuhV1 commented:
EDIT:
Updated upto and and including the StoredData update.
Didn't include the code cleanup (Removal of tabs at end of lines and removal of brackets).

MeuhV1 commented:
Updated to latest version.

dguenms commented on Assets/Python/RFCUtils.py:
```diff
@@ -998,25 +968,13 @@ def getTurns(self, turns): # edead
-		gc.getMap().plot(123, 2).setRevealed(iCiv, False, True, -1);
+			catapult = plotZero.getUnit(0)
+			catapult.kill(False, iCiv)
+		for x in range(-2,3):
```
There also is a surroundingPlots() method in RFCUtils.py now, which should be used for cases like this. It doesn't use module arithmetic yet (i.e. it would cut off on the map edges for the x axis), but that could be fixed/extended.

dguenms commented on Assets/Python/RFCUtils.py:
```diff
@@ -550,31 +516,31 @@ def convertPlotCulture(self, plot, iPlayer, iPercent, bOwner):
-##				pCurrent.setCulture(iLoopCiv, iLoopCivCulture*(100-iPercent)/100, True)
-##				pCurrent.setCulture(iCiv, iCivCulture + iLoopCivCulture*iPercent/100, True)
-		#iCivCulture = pCurrent.getCulture(iCiv)
+##				iLoopCivCulture = plot.getCulture(iLoopCiv)
```
I know this code is commented out, but why is it inconsistently `plot` and `pPlot`?

MeuhV1 replied:
That is because argument of the function this code is part of named it plot. Everything in the function itself is called plot, but in the commented section it was still called pCurrent. I only synchronized the commented section with the rest of the function.

dguenms replied:
Okay, that makes sense.

dguenms commented on Assets/Python/RFCUtils.py:
```diff
@@ -382,11 +350,9 @@ def flipCity(self, tCityPlot, bFlipType, bKillUnits, iNewOwner, iOldOwners):
-							city.setPopulation(2)
+					if bFlipType: #conquest
+						if city.getPopulation() <= 2:
+							city.setPopulation(city.getPopulation() + 1)
```
We also have CvCity::changePopulation(iChange).

dguenms commented on Assets/Python/RFCUtils.py:
```diff
@@ -340,23 +321,10 @@ def flipUnitsInArea(self, lPlots, iNewOwner, iOldOwner, bSkipPlotCity, bKillSett
-							gc.getMap().plot(123, 65).setRevealed(iCiv, False, True, -1);
-			
+						if not bRevealedZero:
+							for x in range(-2,3):
```
Same here, would be cool to use utils.surroundingPlots() here and extend it to wrap around the x axis.

dguenms commented on Assets/Python/RFCUtils.py:
```diff
@@ -255,26 +248,14 @@ def flipUnitsInCityAfter(self, tCityPlot, iCiv):
-		gc.getMap().plot(122, 66).setRevealed(iCiv, False, True, -1);
-		gc.getMap().plot(122, 65).setRevealed(iCiv, False, True, -1);
-		gc.getMap().plot(123, 65).setRevealed(iCiv, False, True, -1);
+		for x in range(-2,3):
```
See below.

dguenms commented on Assets/Python/RFCUtils.py:
```diff
@@ -46,14 +46,11 @@ def getGoalsColor(self, iPlayer): #by CyberChrist
-			cityList.append(pCity.GetCy())
-		if (len(cityList)):	   
+		citylist = self.getCityList(iPlayer)
+		if len(cityList):
```
To check if a list is empty, you can also simply do `if cityList:`.

dguenms commented on Assets/Python/Plague.py:
```diff
@@ -411,51 +378,46 @@ def isVulnerable(self, iPlayer):
-					otherCity = pCity.GetCy()
-					if (otherCity.getX() != city.getX() or otherCity.getY() != city.getY()): #because the city razed still has the plague
+				for otherCity in utils.getCityList(iNewOwner):
+					if city.getID() != otherCity.getID(): #because the city razed still has the plague
```
I think Rhye is checking for a new city on the same plot here. When a city is conquered or changes owners in another way, it has a different ID afterwards iirc.

dguenms commented on Assets/Python/Plague.py:
```diff
@@ -243,40 +228,34 @@ def processPlague(self, iPlayer):
-
-
-	def killUnitsByPlague(self, city, pCurrent, baseValue, iDamage, iPreserveDefenders):
+		if len(lNotInfectedCities):
```
`if lNotInfectedCities` also works here.

dguenms commented on Assets/Python/Plague.py:
```diff
@@ -243,40 +228,34 @@ def processPlague(self, iPlayer):
-
-	def killUnitsByPlague(self, city, pCurrent, baseValue, iDamage, iPreserveDefenders):
+		if len(lNotInfectedCities):
+			if sd.getPlagueCountdown(iPlayer) > 2: #don't spread the last turns
```
Both sd and getPlagueCountdown() shouldn't exist anymore.

dguenms commented on Assets/Python/Plague.py:
```diff
@@ -243,40 +228,34 @@ def processPlague(self, iPlayer):
 											self.infectCity(loopCity)
 											#print ("infect by trade route", city.getName(), "to", loopCity.getName())
+											iHuman = utils.getHumanID()
+											if (gc.getPlayer(iHuman).canContact(iOwner) and iHuman != iOwner):
```
Maybe add an `loopCity.plot().isRevealed(iHuman)` here too? Would avoid the interface clutter with undiscovered cities.

dguenms commented on Assets/Python/Plague.py:
```diff
@@ -170,68 +164,59 @@ def processPlague(self, iPlayer):
+									gc.getTeam(gc.getPlayer(iLoopCiv).getTeam()).isVassal(iPlayer)):
+									if (gc.getPlayer(iLoopCiv).getNumCities() > 0): #this check is needed, otherwise game crashes
+										capital = gc.getPlayer(iLoopCiv).getCapitalCity()
+										if (self.isVulnerable(iLoopCiv) == True):
```
`self.isVulnerable(iLoopCiv)` is fine, no need to check for boolean equality.

dguenms reviewed (changes requested):
Great refactoring, some minor stuff could be improved even further, would be nice if you could include that as well.

Commit:
Improved some code in Plague.py and RFCUtils.py
Code cleanup (removal of tabs at the end of lines, replacing spaces with tabs etc)

MeuhV1 commented:
Updated the PR with your suggestions.

I didn't touch the surroundingPlots function. For unrevealing the plots, the function should return all tiles within the selected radius from the plot. But now, the function only returns plots that are the radius away from the input plot. (Ergo, if the radius is 2, it returns the plots in the second ring of the the input plot, but not the first ring)

dguenms commented:
Okay, seems like I need to take another look at the surroundingPlots() function. Ready to merge otherwise.
